crates/ruff_linter/src/rules/ruff/rules/snapshots/ruff_linter__rules__ruff__rules__unreachable__tests__try.py.md.snap

```diff
@@ -0,0 +1,494 @@
+  block1 --> block0
+  block0 --> return
+```
+
+## Function 6
+### Source
+```python
+def func():
+    try:
+        return 0
+    except:
+        return 1
+    finally:
+        return 2
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1["return 2\n"]
+  block2["return 0\n"]
+  block3[["Exception raised"]]
+  block4["return 1\n"]
+  block5["try:
+        return 0
+    except:
+        return 1
+    finally:
+        return 2\n"]
+
+  start --> block5
+  block5 -- "Exception raised" --> block4
+  block5 -- "else" --> block2
+  block4 --> block1
+  block3 --> return
+  block2 --> block1
+  block1 --> return
+  block0 --> return
+```
+
+## Function 7
+### Source
+```python
+def func():
+    try:
+        raise Exception()
+    except:
+        print("reached")
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1[["Exception raised"]]
+  block2["raise Exception()\n"]
+  block3["print(#quot;reached#quot;)\n"]
+  block4["try:
+        raise Exception()
+    except:
+        print(#quot;reached#quot;)\n"]
+
+  start --> block4
+  block4 -- "Exception raised" --> block3
+  block4 -- "else" --> block2
+  block3 --> block0
+  block2 --> block3
+  block1 --> return
+  block0 --> return
+```
+
+## Function 8
+### Source
+```python
+def func():
+    try:
+        assert False
+        print("unreachable")
+    except:
+        print("reached")
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1["print(#quot;unreachable#quot;)\n"]
+  block2[["Exception raised"]]
+  block3["assert False\n"]
+  block4["print(#quot;reached#quot;)\n"]
+  block5["try:
+        assert False
+        print(#quot;unreachable#quot;)
+    except:
+        print(#quot;reached#quot;)\n"]
+
+  start --> block5
+  block5 -- "Exception raised" --> block4
+  block5 -- "else" --> block3
+  block4 --> block0
+  block3 -- "False" --> block1
+  block3 -- "else" --> block4
+  block2 --> return
+  block1 --> block0
+  block0 --> return
+```
+
+## Function 9
+### Source
+```python
+def func():
+    try:
+        raise Exception()
+    finally:
+        print('reached')
+        return 2
+```
+
+### Control Flow Graph
+```mermaid
+flowchart TD
+  start(("Start"))
+  return(("End"))
+  block0[["`*(empty)*`"]]
+  block1["print('reached')\nreturn 2\n"]
+  block2[["Exception raised"]]
+  block3["raise Exception()\n"]
+  block4["try:
+        raise Exception()
+    finally:
+        print('reached')
+        return 2\n"]
+
+  start --> block4
+  block4 -- "Exception raised" --> block2
+  block4 -- "else" --> block3
+  block3 --> block1
+  block2 --> return
+  block1 --> return
+  block0 --> return
+```
+
+## Function 10
+### Source
+```python
+def func():
+    try:
+        assert False
+        print("unreachable")
+    finally:
+        print("reached")
+```
```

Comment:
We should add some tests where we raise exceptions inside blocks other than `try`. E.g.

```python
try:
    assert False
except ex:
    raise ex

finally:
    raise Exception("other")
```

Ideally we combine this with a nested `try` block.

Reply:
I added a big one, and there were issues, so good catch